### ISSUE TITLE
getItem and setItem implementation for localstorage

### ISSUE THREAD

**Issue by seetdev** [clarkbw/jest-localstorage-mock#110]
Hi,

Thanks for the nice work in creating this. I was working with it and I am wondering it both the setItem and getItem should force the value to become a string. Looking back into the specs. The interface shows that the values should be DOMString. I am not sure if I am reading the DOMString spec right but that appears to be a string as well. Mozilla indicates it to be a string as well.

I am wondering if there might be code required to handle JSON values as well. I tried accessing the store for github and I think the value is converted to be a string on Chrome and Firefox when the value is a json object

https://github.com/clarkbw/jest-localstorage-mock/blob/50f3dfb83091f789c59f4fc2f61b5ffc76ad69b6/src/localstorage.js#L3-L13

Image of localStorage.getItem in Chrome and Mozilla for json values
![Screen Shot 2020-01-06 at 1 04 03 PM](https://user-images.githubusercontent.com/35116035/71837777-3703c900-3085-11ea-9802-e477a5e71ec3.png)

Image of interface for localStorage
![Screen Shot 2020-01-06 at 12 52 29 PM](https://user-images.githubusercontent.com/35116035/71837052-7f21ec00-3083-11ea-80f1-6bb9b6132aa0.png)
Reference: https://html.spec.whatwg.org/multipage/webstorage.html

DOMString: 
https://heycam.github.io/webidl/#idl-DOMString
https://developer.mozilla.org/en-US/docs/Web/API/DOMString

**Comment by rodrigograca31** [comment]
I'm having a similar issue.

after doing `localStorage.setItem('token', null);` or trying to get a key that doesnt exist like `localStorage.getItem('non-existing-key');` it returns `"null"` as a string.

this causes problems because my code is `if (!token)` which in the browser is `if(!null)` and evaluates to `true` but here is `if(!"null")` and evaluates to `false`........


(also I remove this plugin and was still getting the same functionality/problem.... I wonder what was up with that.......... 🤔)

**Comment by seetdev** [comment]
Actually if you do "localStorage.setItem('token', null)" and "localStorage.getItem('token')" following that it should give you "null". That is actually the correct behavior according to the W3C specifications for localStorage. The image below shows the behavior in Safari, Firefox and Chrome. So you might want to check your code's behavior in the browsers. The mock does not follow the behavior of the specifications. Since it stores and returns the values as the type that you supplies to it 

![Screen Shot 2020-03-05 at 10 36 26 AM](https://user-images.githubusercontent.com/35116035/75997501-6ab86f00-5ecd-11ea-9585-f9fa24c563e4.png)

**Comment by rodrigograca31** [comment]
True....! and now I can't replicate my problem on my tests..... not sure why..... :thinking: 
maybe I got confused yesterday... sorry :pray:
